### DISCUSSION TITLE
Sunton ESP32-8048S070 - is it compatible?

### DISCUSSION THREAD

**Discussion by themagicm** [lovyan03/LovyanGFX#350] [Q&A]
Hello..
Is it possible to use the below display with this library?  If so, is there sample code or any settings somewhere I can use to utilize this library?

https://www.makerfabs.com/sunton-esp32-s3-7-inch-tn-display-with-touch.html

**Comment by lovyan03** [comment]
Hello, @themagicm 

Please refer to this setup.
https://github.com/lovyan03/LovyanGFX/blob/master/src/lgfx_user/LGFX_ESP32S3_RGB_ESP32-8048S043.h

**Reply by themagicm** [reply]
Interesting... I was thinking that I'd try that to see what happens.   Thanks, I'll let you know.

**Reply by bastelpingu** [reply]
hi @lovyan03 

I have the display at home and have just tried the code but unfortunately it is displayed all wrong. 

like here: 

https://github.com/lovyan03/LovyanGFX/discussions/350#discussioncomment-6055090

do you have any idea what it can be?

thank you!

**Comment by Radek-Suski** [comment]
Any luck with that? 

I am only getting something like this: 

https://github.com/lovyan03/LovyanGFX/assets/1110663/0d225b0b-fb0f-4a93-908b-c86fbb3bfa8f

**Reply by bastelpingu** [reply]
with me it looks exactly the same... have you been able to fix it somehow?

**Reply by Radek-Suski** [reply]
Not yet. I am trying to start from the bottom with pure ESP32-IDF and see how it goes

**Comment by TangerineDreamer** [comment]
Hi, Radek
i've got the same result,even if i put all the settings of the arduino-gfx setup that works well ,in the lgfx setup,
but strangely the values working with arduino-gfx doesn't fit the datacheet from makerfabs.

but i think i have strange "better results" if i use this 3.5 inch setup coming from there:

https://github.com/Makerfabs/Makerfabs-ESP32-S3-SPI-TFT-with-Touch,

i know it's dummy because it's not 16 bits parrallel
i can see for a very short time some geometrics or rectangles.
they preconise PSRAM "disabled",but "OPI" works fine.

**Reply by Radek-Suski** [reply]
Thanks. Will check. But to be honest with arduino-gfx I have also [weird result ](https://github.com/moononournation/Arduino_GFX/discussions/299)

**Comment by MMlodzinski** [comment]
Hello everyone,

I've made an example using pure ESP-IDF (5.1 version) that make use of the screen and touch of this board. Maybe it will help you somehow, link to repo below:
[Sunton_ESP32-8048S070c LVGL example](https://github.com/MMlodzinski/Sunton_ESP32-8048S070c_ESP_IDF_LVGL_example)

**Comment by TangerineDreamer** [comment]
Good thing,can you build from this example, a lovyan gfx setup for us?

**Reply by MMlodzinski** [reply]
Well, to be honest I've never used lovyan gfx. So I could try, but don't expect results very soon, as I don't have much time.

**Comment by TangerineDreamer** [comment]
We have time :)
A bunch of guys are waiting for !!

**Comment by mhaberler** [comment]
here's an Arduino, non-lovyanGFX example for the 7in display: https://github.com/mhaberler/squareline-esp32/blob/8b614edb3e3b8ee7108dd8737460bb2105722816/src/sunton/main.cpp to take "inspiration" from ;)

**Comment by TLaren** [comment]
Long time no action.
If anyone is watching...

Using lovyanGFX 1.1.12
Using the example I found the following:

The recommended [LGFX_ESP32S3_RGB_ESP32-8048S043.h](https://github.com/lovyan03/LovyanGFX/blob/master/src/lgfx_user/LGFX_ESP32S3_RGB_ESP32-8048S043.h) file is for the 4.3 display.
Most of the pins are wrong. So here is the ones that worked for me.
```
#include "hal/gpio_types.h"

#define LGFX_USE_V1
#include <LovyanGFX.hpp>

#include <lgfx/v1/platforms/esp32s3/Panel_RGB.hpp>
#include <lgfx/v1/platforms/esp32s3/Bus_RGB.hpp>

#include <driver/i2c.h>

class LGFX : public lgfx::LGFX_Device
{
public:

  lgfx::Bus_RGB     _bus_instance;
  lgfx::Panel_RGB   _panel_instance;
  lgfx::Light_PWM   _light_instance;
  lgfx::Touch_GT911 _touch_instance;

  LGFX(void)
  {
    {
      auto cfg = _panel_instance.config();

      cfg.memory_width  = 800;
      cfg.memory_height = 480;
      cfg.panel_width  = 800;
      cfg.panel_height = 480;

      cfg.offset_x = 0;
      cfg.offset_y = 0;

      _panel_instance.config(cfg);
    }

    {
      auto cfg = _panel_instance.config_detail();

      cfg.use_psram = 1;

      _panel_instance.config_detail(cfg);
    }

    {
      auto cfg = _bus_instance.config();
      cfg.panel = &_panel_instance;
      cfg.pin_d0  = GPIO_NUM_15;  // B0
      cfg.pin_d1  = GPIO_NUM_7;  // B1
      cfg.pin_d2  = GPIO_NUM_6; // B2
      cfg.pin_d3  = GPIO_NUM_5;  // B3
      cfg.pin_d4  = GPIO_NUM_4;  // B4
      cfg.pin_d5  = GPIO_NUM_9;  // G0
      cfg.pin_d6  = GPIO_NUM_46;  // G1
      cfg.pin_d7  = GPIO_NUM_3;  // G2
      cfg.pin_d8  = GPIO_NUM_8; // G3
      cfg.pin_d9  = GPIO_NUM_16; // G4
      cfg.pin_d10 = GPIO_NUM_1;  // G5
      cfg.pin_d11 = GPIO_NUM_14; // R0
      cfg.pin_d12 = GPIO_NUM_21; // R1
      cfg.pin_d13 = GPIO_NUM_47; // R2
      cfg.pin_d14 = GPIO_NUM_48; // R3
      cfg.pin_d15 = GPIO_NUM_45; // R4

      cfg.pin_henable = GPIO_NUM_41;
      cfg.pin_vsync   = GPIO_NUM_40;
      cfg.pin_hsync   = GPIO_NUM_39;
      cfg.pin_pclk    = GPIO_NUM_42;
      cfg.freq_write  = 14000000;

      cfg.hsync_polarity    = 0;
      cfg.hsync_front_porch = 80;
      cfg.hsync_pulse_width = 4;
      cfg.hsync_back_porch  = 16;
      cfg.vsync_polarity    = 0;
      cfg.vsync_front_porch = 22;
      cfg.vsync_pulse_width = 13;
      cfg.vsync_back_porch  = 10;
      cfg.pclk_idle_high    = 1;
      _bus_instance.config(cfg);
    }
    _panel_instance.setBus(&_bus_instance);

    {
      auto cfg = _light_instance.config();
      cfg.pin_bl = GPIO_NUM_2;
      _light_instance.config(cfg);
    }
    _panel_instance.light(&_light_instance);

    {
      auto cfg = _touch_instance.config();
      cfg.x_min      = 0;
      cfg.x_max      = 800;
      cfg.y_min      = 0;
      cfg.y_max      = 480;
      cfg.pin_int    = GPIO_NUM_NC;
      cfg.bus_shared = false;
      cfg.offset_rotation = 1;
      // I2C接続
      cfg.i2c_port   = I2C_NUM_1;
      cfg.pin_sda    = GPIO_NUM_19;
      cfg.pin_scl    = GPIO_NUM_20;
      cfg.pin_rst    = GPIO_NUM_38;
      cfg.freq       = 400000;
      cfg.i2c_addr   = 0x14;        // 0x5D , 0x14
      _touch_instance.config(cfg);
      _panel_instance.setTouch(&_touch_instance);
    }

    setPanel(&_panel_instance);
  }
};
```
I created a file LGFX_ESP32S3_RGB_ESP32-8049S070.h
and put in the folder then included it
#include "LGFX_ESP32S3_RGB_ESP32-8049S070.h"

Commented out the autodetect files
//#define LGFX_AUTODETECT // 自動認識 (D-duino-32 XS, WT32-SC01, PyBadge はパネルID読取りが出来ないため自動認識の対象から外れています)
//#include <LGFX_AUTODETECT.hpp>  // クラス"LGFX"を準備します
 and it compiles and runs.
The file I used was example 1_simple_use_LGFX.ino

I hope that someone finds this useful.
Like it or comment.

**Reply by TLaren** [reply]
After further investigation I found that in lgfx_user there is LGFX_Sunton_ESP32-8048S070.h that works great.
Just add #include <lgfx_user/LGFX_Sunton_ESP32-8048S070.h> just after #include <LovyanGFX.hpp>

So the first few lines of your program should be something like this:

#include <LovyanGFX.hpp>
#include <lgfx_user/LGFX_Sunton_ESP32-8048S070.h>

static LGFX lcd;

**Comment by TangerineDreamer** [comment]
Hi,TLaren
THANK YOU!!!
Sure i am not alone to say that,many guys struggle to make it work and quit,but you did it :)
All the examples are now working for me,feel better today :)

**Comment by TangerineDreamer** [comment]
Hey Tobozo,
The "Sunton ESP32-8048S070" setup found by TLaren is working,can you put it in the "lgfx_user" ?
And why not write  the title header of the present post as [solved] ?

**Reply by TLaren** [reply]
Actually it is there, well now, I found it a day or so ago. See my comment above.

**Comment by TangerineDreamer** [comment]
Ooops sorry ,it's my bad,
It has been here for 10 months and i did'nt noticed it   :)
That's the reason i think it's better to change the title of this present post --> [solved]
Maybe tobozo can do it.
